src/libbson/src/bson/bson-atomic.h

```diff
@@ -453,6 +454,9 @@ bson_atomic_thread_fence ()
    BSON_IF_GNU_LIKE (__sync_synchronize ();)
 }
 
+BSON_GNUC_DEPRECATED_FOR ("bson_atomic_thread_fench")
+BSON_EXPORT (void) bson_memory_barrier (void);
```

Comment:
```suggestion
BSON_GNUC_DEPRECATED_FOR ("bson_atomic_thread_fence")
BSON_EXPORT (void) bson_memory_barrier (void);
```

Reply:
oops.

Also, my only purpose with those tests was only to least to get the compiler's eyes on each control flow path. I'm not even sure if it is possible to verify atomics are correct short of just having a knowledgeable expert stare at the code and generated assembly.